From 5136fe378564ec3c9b0f0cf2aac81e4dd102c638 Mon Sep 17 00:00:00 2001
From: Vaibhav Agarwal <vaibhav.agarwal@intel.com>
Date: Fri, 22 Dec 2017 01:10:48 +0800
Subject: [PATCH 0858/2302] drm/i915: Fix array index out of bound access issue

"Buffer Overflow - Array Index Out of Bounds, intel_pm.c: 3093" error is
reported during static code analysis

Since, the max level is increased to 7 after foll. patch, commit
2af30a5c40bf
("drm/i915/skl: Read the Memory Latency Values for WM computation") so,
element_count for array maintaining each wm_level should also be
modified.

Signed-off-by: Vaibhav Agarwal <vaibhav.agarwal@intel.com>
Reviewed-by: Pankaj Laxminarayan Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
---
 drivers/gpu/drm/i915/intel_drv.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_drv.h b/drivers/gpu/drm/i915/intel_drv.h
index 3adb9c3..73b1b59 100644
--- a/drivers/gpu/drm/i915/intel_drv.h
+++ b/drivers/gpu/drm/i915/intel_drv.h
@@ -495,7 +495,7 @@ struct intel_crtc_scaler_state {
 #define I915_MODE_FLAG_INHERITED 1
 
 struct intel_pipe_wm {
-	struct intel_wm_level wm[5];
+	struct intel_wm_level wm[7];
 	uint32_t linetime;
 	bool fbc_wm_enabled;
 	bool pipe_enabled;
-- 
2.7.4

